jobs/mongodb_migration/src/mongodb_migration/migrations/_20230309141600_cache_add_job_runner_version.py

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+
+from libcommon.simple_cache import CachedResponse
+from mongoengine.connection import get_db
+
+from mongodb_migration.check import check_documents
+from mongodb_migration.migration import Migration
+
+
+# connection already occurred in the main.py (caveat: we use globals)
+class MigrationAddJobRunnerVerionToCacheResponse(Migration):
+    def up(self) -> None:
+        # See https://docs.mongoengine.org/guide/migration.html#example-1-addition-of-a-field
+        logging.info("If missing, add 'job_runner_version' field based on 'worker_version' value")
+        db = get_db("cache")
+        db["cachedResponsesBlue"].update_many(
+            {"job_runner_version": {"$exists": False}},
+            [
+                {"$set": {"job_runner_version": {"$toInt": {"$first": {"$split": ["$worker_version", "."]}}}}},
+                {"$unset": "worker_version"},
+            ],  # type: ignore
+        )
+
+    def down(self) -> None:
+        logging.info("Remove 'job_runner_version' field from all the cached results")
+        db = get_db("cache")
+        db["cachedResponsesBlue"].update_many({}, {"$unset": {"job_runner_version": ""}})
```

Comment:
Here, worker_version is lost. I think we should really split this migration in two:
- a Migration that creates and populates the job_runner_version
- the, an IrreversibleMigration that deletes worker_version (see https://github.com/huggingface/datasets-server/blob/1e5064e2ef711741bdf7ec579898ce63495a5c43/jobs/mongodb_migration/tests/test_plan.py#L71)